cpp/open3d/geometry/TriangleMesh.cpp

```diff
@@ -1188,6 +1188,36 @@ double TriangleMesh::GetSurfaceArea(std::vector<double> &triangle_areas) const {
     return surface_area;
 }
 
+double TriangleMesh::GetVolume() const {
+    if (!IsWatertight()) {
+        utility::LogError(
+                "The mesh is not watertight, and the volume cannot be "
+                "computed.");
+    }
+
+    double volume = 0;
+    for (size_t tidx = 0; tidx < triangles_.size(); ++tidx) {
```

Comment:
This code block is easily parallelizable, and I guess OpenMP could be used here. Are the following lines all that's needed?
```
#ifdef _OPENMP
#pragma omp parallel for schedule(static)
#endif
```

Reply:
Yes, that's all we need, with two catches:

- For reduction, you may need `#pragma omp parallel for reduction ....`
- Certain MSVC compilers require signed indices (e.g. `int` or `int64_t`). If `size_t` fails on our Windows CI, you may need to change to signed indices.